cou/utils/app_utils.py

```diff
@@ -126,3 +109,98 @@ def validate_ovn_support(version: str) -> None:
+    logger.debug("Running '%s' on '%s'", check_command, unit)
+
+    try:
+        check_option_result = await model.run_on_unit(
+            unit_name=unit, command=check_command, timeout=600
+        )
+        if str(check_option_result["Code"]) == "0":
+            logger.debug(check_option_result["Stdout"])
+
+            current_require_osd_release = json.loads(check_option_result["Stdout"]).get(
+                "require_osd_release", ""
+            )
+            logger.debug("Current require-osd-release is set to: %s", current_require_osd_release)
+        else:
+            raise RunUpgradeError(
+                "Cannot determine the current value of "
+                f"require_osd_release on ceph-mon unit '{unit}'."
+            ) from CommandRunFailed(cmd=check_command, result=check_option_result)
+
+    except JujuError as exc:
+        raise RunUpgradeError(
+            f"Cannot determine the current value of require_osd_release on ceph-mon unit '{unit}'."
+        ) from exc
+
+    return current_require_osd_release
+
+
+async def _get_current_osd_release(unit: str, model: COUModel) -> str:
+    """Get the current release of OSDs.
+
+    The release of OSDs is parsed from the output of running `ceph versions` command
+    on a ceph-mon unit.
+    :param unit: The ceph-mon unit name where the check command runs on.
+    :type unit: str
+    :param model: COUModel object
+    :type model: COUModel
+    :return: the release which OSDs are on
+    :rtype: str
+    :raises RunUpgradeError: When an upgrade fails.
+    """
+    check_command = "ceph versions -f json"
+    logger.debug("Running '%s' on '%s'", check_command, unit)
+
+    try:
+        check_osd_result = await model.run_on_unit(
+            unit_name=unit, command=check_command, timeout=600
+        )
+        if str(check_osd_result["Code"]) == "0":
+            logger.debug(check_osd_result["Stdout"])
+
+            osd_release_output = json.loads(check_osd_result["Stdout"]).get("osd", None)
+            # throw exception if ceph-mon doesn't contain osd release information in `ceph`
+            if not osd_release_output:
+                raise RunUpgradeError(
+                    f"Cannot get OSD release information on ceph-mon unit '{unit}'."
+                )
+            # throw exception if OSDs are on mismatched releases
+            if len(osd_release_output) > 1:
+                raise RunUpgradeError(
+                    f"OSDs are on mismatched releases:\n{osd_release_output}."
+                    "Please manually upgrade them to be on the same release before proceeding."
+                )
+
+            # get release name from "osd_release_output". Example value of "osd_release_output":
+            # {'ceph version 15.2.17 (8a82819d84cf884bd39c17e3236e0632) octopus (stable)': 1}
+            osd_release_key, *_ = osd_release_output.keys()
+            current_osd_release = osd_release_key.split(" ")[4].strip()
```

Comment:
I still think that a dictionary on openstack.py having the ceph  could be beneficial here.

```python
CEPH_RELEASES = OrderedDict(
    [
        ....
        ("pacific", VersionRange("16.2.0", "17.2.0" )),
        .....
    ]
)

....
if current_osd_release not in CEPH_RELEASES.keys():
     raise RunUpgradeError(f"Invalid ceph-osd release '{current_osd_release}'")
```